### DISCUSSION TITLE
Some internal function conflict with user define function name "delete"

### DISCUSSION THREAD

**Discussion by emptizen** [alpinejs/alpine#4436] [7. Bugs (archived)]
When the function of x-data is named `delete`, Alpine will report the following error, for detailed stack trace please see here https://jsfiddle.net/ntve9org/2/. If I changed it to another name, everything was fine.
I don't know if there are any internal functions/properties that might conflict with user-defined function names,

```js
"Alpine Expression Error: Unexpected token '}'
```
Please refer https://jsfiddle.net/ntve9org/2/  to see the full detail.

**Comment by emptizen** [comment]
I am new to Javascript as a backend developer, surprised that **delete** is reserved word in Javascript! 😰

**Reply by ekwoka** [reply]
Yeah it's used for deleting a property off of an object. So that it won't be there at all.

It's the only way to mutate an object to remove a key.
